internal/consensus/context.go

```diff
@@ -35,3 +36,17 @@ func msgInfoFromCtx(ctx context.Context) msgInfo {
 	val := ctx.Value(msgInfoCtx)
 	return val.(msgInfo)
 }
+
+// ctxWithLogKeyVals puts key-vals log slice into the context
+func ctxWithLogKeyVals(ctx context.Context, keyVals []any) context.Context {
```

Comment:
I didn't use `logger.With()` because of 
1. `With` returns a copy of the logger with the passed parameters
2. I remember we had a performance issue with this in the past

btw, I like the approach when a logger can present in a context, it can help us to make a code simpler, in terms of dependencies.